src/main/java/com/adyen/Client.java

```diff
@@ -87,23 +89,52 @@ public Client(String apiKey, Environment environment, int connectionTimeoutMilli
         this.config.setConnectionTimeoutMillis(connectionTimeoutMillis);
     }
 
+    public Client(String username, String password, Environment environment, int connectionTimeoutMillis, String liveEndpointUrlPrefix) {
+
+        this.config = new Config();
+        this.config.setUsername(username);
+        this.config.setPassword(password);
+        this.setEnvironment(environment, liveEndpointUrlPrefix);
+        this.config.setConnectionTimeoutMillis(connectionTimeoutMillis);
+    }
+
+    public Client(String apiKey, Environment environment, int connectionTimeoutMillis, String liveEndpointUrlPrefix) {
+
+        this.config = new Config();
+        this.config.setApiKey(apiKey);
+        this.setEnvironment(environment, liveEndpointUrlPrefix);
+        this.config.setConnectionTimeoutMillis(connectionTimeoutMillis);
+    }
+
+    /**
+     * @deprecated As of library version 1.5.4, replaced by {@link #setEnvironment(Environment environment, String liveEndpointUrlPrefix)}.
+     */
     public void setEnvironment(Environment environment) {
+        this.setEnvironment(environment, null);
+    }
 
-        if (environment.equals(Environment.TEST)) {
+    /**
+     * @param environment
+     * @param liveEndpointUrlPrefix - Provide the [random]-[company name] combination from the "API URLs and Response" menu in the Adyen Customer Area
+     */
+    public void setEnvironment(Environment environment, String liveEndpointUrlPrefix) {
+
+        if (Environment.TEST.equals(environment)) {
             this.config.setEnvironment(environment);
             this.config.setEndpoint(ENDPOINT_TEST);
             this.config.setMarketPayEndpoint(MARKETPAY_ENDPOINT_TEST);
             this.config.setHppEndpoint(HPP_TEST);
             this.config.setCheckoutEndpoint(CHECKOUT_ENDPOINT_TEST);
-
-        } else if (environment.equals(Environment.LIVE)) {
+        } else if (Environment.LIVE.equals(environment)) {
             this.config.setEnvironment(environment);
-            this.config.setEndpoint(ENDPOINT_LIVE);
             this.config.setMarketPayEndpoint(MARKETPAY_ENDPOINT_LIVE);
             this.config.setHppEndpoint(HPP_LIVE);
-            this.config.setCheckoutEndpoint(CHECKOUT_ENDPOINT_LIVE);
-        } else {
-            // throw exception
+            if (liveEndpointUrlPrefix != null && ! liveEndpointUrlPrefix.isEmpty()) {
+                this.config.setEndpoint(ENDPOINT_PROTOCOL + liveEndpointUrlPrefix + ENDPOINT_LIVE_SUFFIX);
+                this.config.setCheckoutEndpoint(ENDPOINT_PROTOCOL + liveEndpointUrlPrefix + CHECKOUT_ENDPOINT_LIVE_SUFFIX);
+            } else {
+                this.config.setEndpoint(ENDPOINT_LIVE);
+            }
```

Comment:
Maybe best to setCheckoutEndpoint to null so if you first do a setEnvironment(TEST) and then setEnvironment(LIVE) you don't use the test endpoint and our exception can be triggered. Now it will ignore it as it has a value.